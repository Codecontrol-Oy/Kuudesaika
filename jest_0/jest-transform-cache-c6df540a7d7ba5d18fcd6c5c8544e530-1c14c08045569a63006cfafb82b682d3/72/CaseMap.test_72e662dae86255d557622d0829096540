'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _CaseMap = require('./CaseMap');var _CaseMap2 = _interopRequireDefault(_CaseMap);
var _Case = require('../../../mock/Case.json');var _Case2 = _interopRequireDefault(_Case);
var _Actions = require('../../../mock/Actions.json');var _Actions2 = _interopRequireDefault(_Actions);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('<CaseMap />', function () {

    it('Should match snapshot', function () {
        var caseMap = shallow(
        _react2.default.createElement(_CaseMap2.default, { currentCase: _Case2.default.cases[0], actions: _Actions2.default.actions }));

        expect(caseMap).toMatchSnapshot();
    });
});