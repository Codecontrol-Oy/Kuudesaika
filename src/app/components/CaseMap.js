import React from 'react';
import PropTypes from 'prop-types';
import Map from 'grommet/components/Map';
import Card from 'grommet/components/Card';
import Box from 'grommet/components/Box';
import Section from 'grommet/components/Section';
import Article from 'grommet/components/Article';
import Heading from 'grommet/components/Heading';
import Anchor from 'grommet/components/Anchor';
import {browserHistory} from 'react-router';
import {getCity} from 'Actions';
import CheckMarkIcon from 'grommet/components/icons/base/Checkmark';
import InformationIcon from 'grommet/components/icons/base/CircleInformation';
import ArchivedIcon from 'grommet/components/icons/base/Archive';
import RejectedIcon from 'grommet/components/icons/base/Close';
import ReturnedIcon from 'grommet/components/icons/base/Cycle';
import Timestamp from 'grommet/components/Timestamp';
import theme from '../theme/global.scss';
export default class CaseMap extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
        selectedOrganization: null
    };
  }

  openOrganization = (id) => {
   if (id) {
        browserHistory.push(`/${getCity()}/organisaatio/` + id);
        window.location.reload();
   }
  }

  selectOrganization = (index) => {
      if (this.state.selectedOrganization !== index) {
        this.setState({selectedOrganization: index });
      }
  }

  getResolution = (value) => {
        switch (value) {
            case 'PASSED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty ehdotuksen mukaan</Heading>;
            }
            case 'PASSED_MODIFIED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty esittelijän ehdotuksesta poiketen</Heading>;
            }
            case 'PASSED_REVISED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty esittelijän muutetun ehdotuksen mukaan</Heading>;
            }
            case 'PASSED_VOTED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty ehdotuksen mukaan äänestyksin</Heading>;
            }
            case '': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><InformationIcon colorIndex={"unknown"} /> Ei tietoa päätöksestä</Heading>;
            }
            case 'NOTED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><InformationIcon colorIndex={"unknown"} /> Merkittiin tiedoksi</Heading>;
            }
            case 'TABLED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><ArchivedIcon colorIndex={"unknown"} /> Pöydätty</Heading>;
            }
            case 'REJECTED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><RejectedIcon colorIndex={"critical"} /> Hylätty</Heading>;
            }
            case 'RETURNED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><ReturnedIcon colorIndex={"warning"} /> Palautettu</Heading>;
            }
            default:
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><ReturnedIcon colorIndex={"warning"} />-</Heading>;
        }
  }

  gatherCaseMap = () => {
    const categories = [];
    const links = [];
    categories.push({
            'id': 'category-original',
            'label': 'Asia',
            'items': []
         });
    categories.push({
            'id': 'category-sections',
            'label': 'Käsittelypisteet',
            'items': []
         });
        categories[0].items.push({
                'id': this.props.currentCase.id.toString(),
                'label': this.props.currentCase.register_id,
                'node': <Box colorIndex='neutral-1' pad={"small"}>{this.props.currentCase.register_id}</Box>
                });

    for (let i = 0; i < this.props.actions.length; i++) {

        const isEventBased = this.props.actions[i].event ? true : false;
        const organization = this.props.actions[i].event ? this.props.actions[i].event.organization : this.props.actions[i].post.organization;
        const startDate = isEventBased ? this.props.actions[i].event.start_date : this.props.actions[i].post.start_date;
        categories[1].items.push({
                'id': this.props.actions[i].id.toString(),
                'label': isEventBased ? this.props.actions[i].event.organization.name : this.props.actions[i].post.organization.name,
                'node': <Card colorIndex='neutral-4'
                              pad={"small"}
                              label={organization.classification}
                              heading={organization.name}
                              description={<Section>{startDate && <b><Timestamp fields={"date"} value={startDate}/></b>}<Article>{this.props.actions[i].title}</Article><Article>{this.getResolution(this.props.actions[i].resolution)}</Article></Section>}
                              link={<Anchor onClick={() => this.openOrganization(this.state.selectedOrganization)} label={"siirry organisaatioon"} />}
                              >
                        </Card>
                });
    }
               
    for (let i = 0; i < categories[1].items.length; i++) {
        links.push({'parentId': this.props.currentCase.id.toString(), 'childId': categories[1].items[i].id.toString()});
    }

    return <Map onClick={() => this.openOrganization(this.state.selectedOrganization)} onActive={this.selectOrganization} active={this.state.selectedOrganization} vertical={false} data={{'categories': categories, 'links': links}} />;

  }

  render () {
    const caseMap = this.gatherCaseMap();
    return (
            <Section>
                {caseMap}
            </Section>
    );
  }
}

CaseMap.propTypes = {
    currentCase: PropTypes.object
};
