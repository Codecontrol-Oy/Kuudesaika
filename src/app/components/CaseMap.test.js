import React from 'react';
import CaseMap from './CaseMap';
import case_mock from '../../../mock/Case.json';
import actions_mock from '../../../mock/Actions.json';

describe('<CaseMap />', () => {

    it('Should match snapshot', () => {
        const caseMap = shallow(
            <CaseMap currentCase={case_mock.cases[0]} actions={actions_mock.actions} />
        );
        expect(caseMap).toMatchSnapshot();
    });
});