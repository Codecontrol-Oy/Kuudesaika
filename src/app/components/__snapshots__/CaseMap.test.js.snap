// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaseMap /> Should match snapshot 1`] = `
<Section
  pad={
    Object {
      "vertical": "medium",
    }
  }
>
  <ResourceMap
    active={null}
    data={
      Object {
        "categories": Array [
          Object {
            "id": "category-original",
            "items": Array [
              Object {
                "id": "selected-1",
                "label": "HEL 2012-002954",
                "node": <Box
                  announce={false}
                  colorIndex="neutral-1"
                  direction="column"
                  focusable={true}
                  pad="small"
                  responsive={true}
                  tag="div"
      >
                  HEL 2012-002954
      </Box>,
              },
            ],
            "label": "Asia",
          },
          Object {
            "id": "category-sections",
            "items": Array [
              Object {
                "id": "1",
                "label": "Kaupunginhallitus",
                "node": <Card
                  a11yTitle="Card"
                  colorIndex="neutral-4"
                  contentPad="medium"
                  description={
                              <Section
                                          pad={
                                                      Object {
                                                                  "vertical": "medium",
                                                                }
                                          }
                              >
                                          <b>
                                                      <Timestamp
                                                                  fields="date"
                                                                  value="2013-04-18"
                                                      />
                                          </b>
                                          <Article
                                                      direction="column"
                                                      pad="none"
                                          >
                                                      Kokouksen laillisuuden ja päätösvaltaisuuden toteaminen sekä pöytäkirjan tarkastajien valinta
                                          </Article>
                                          <Article
                                                      direction="column"
                                                      pad="none"
                                          >
                                                      <Breadcrumb
                                                                  resolution="PASSED"
                                                      />
                                          </Article>
                              </Section>
                  }
                  heading="Kaupunginhallitus"
                  headingStrong={true}
                  label="Hallitus"
                  link={
                              <Anchor
                                          animateIcon={true}
                                          label="siirry organisaatioon"
                                          method="push"
                                          onClick={[Function]}
                                          tag="a"
                              />
                  }
                  pad="small"
                  textSize="small"
      />,
              },
              Object {
                "id": "2",
                "label": "Kaupunginhallitus",
                "node": <Card
                  a11yTitle="Card"
                  colorIndex="neutral-4"
                  contentPad="medium"
                  description={
                              <Section
                                          pad={
                                                      Object {
                                                                  "vertical": "medium",
                                                                }
                                          }
                              >
                                          <b>
                                                      <Timestamp
                                                                  fields="date"
                                                                  value="2013-04-18"
                                                      />
                                          </b>
                                          <Article
                                                      direction="column"
                                                      pad="none"
                                          >
                                                      Asuntolautakunnan pöytäkirjojen nähtävänä pitäminen vuonna 2012
                                          </Article>
                                          <Article
                                                      direction="column"
                                                      pad="none"
                                          >
                                                      <Breadcrumb
                                                                  resolution="PASSED"
                                                      />
                                          </Article>
                              </Section>
                  }
                  heading="Kaupunginhallitus"
                  headingStrong={true}
                  label="Hallitus"
                  link={
                              <Anchor
                                          animateIcon={true}
                                          label="siirry organisaatioon"
                                          method="push"
                                          onClick={[Function]}
                                          tag="a"
                              />
                  }
                  pad="small"
                  textSize="small"
      />,
              },
            ],
            "label": "Käsittelypisteet",
          },
        ],
        "links": Array [
          Object {
            "childId": "1",
            "parentId": "selected-1",
          },
          Object {
            "childId": "2",
            "parentId": "selected-1",
          },
        ],
      }
    }
    linkColorIndex="graph-1"
    onActive={[Function]}
    onClick={[Function]}
    vertical={false}
  />
</Section>
`;
